File: yukicoder/FizzBuzz/FizzBuzz.java
import java.util.Scanner;

public class FizzBuzz {
    public static void main(String[] args){
        Scanner S = new Scanner(System.in);
        int N = S.nextInt();
        
        for(int i=1;i<=N;i++){
            if(i%15==0){
                System.out.println("FizzBuzz");
                continue;
            }
            else if(i%5==0){
                System.out.println("Buzz");
                continue;
            }
            else if(i%3==0){
                System.out.println("Fizz");
                continue;
            }
            else
                System.out.println(i);
        }
    }
}

File: yukicoder/FizzBuzz/FizzBuzz.kt
fun Buzz(n : Int) : Boolean = n % 5 == 0

fun FizzBuzz(N : Int){
    val Fizz : (Int) -> Boolean = {(n : Int) -> n % 3 == 0}
//    val Buzz : (Int) -> Boolean = {(n : Int) -> n % 5 == 0}
    for(i in 1..N) {
        println(
                when{
                    Fizz(i) && Buzz(i) -> "FizzBuzz"
                    Fizz(i) -> "Fizz"
                    Buzz(i) -> "Buzz"
                    else -> i
                }
        )
    }
}

fun main(args : Array<String>){
    FizzBuzz(readLine()!!.toInt())
}
